specs/eid-hermes-commands-format.md

```diff
@@ -0,0 +1,227 @@
+# eid-hermes Commands
+
+## Introduction
+
+This document defines the commands that can be sent to a Hermes device. It also
+defines the command completions which will be sent by the device back to the
+host. It includes both the commands' format as well as which commands are
+available.
+
+## Sample Execution
+
+The following diagram exemplifies a possible execution flow for Hermes. Note
+how the data slot is reused to process data in chunks at a time.
+
+```
+┌--------------┐     ┌---------------┐    ┌--------------┐     ┌------------┐
+| Request slot | --> | Write program | -->| Request slot | --> | Write data |<-┐
+|   (program)  |     └---------------┘    |    (data)    |     └------------┘  |
+└--------------┘                          └--------------┘             |       |
+                                                                       v       |
+┌--------------┐     ┌--------------┐     ┌--------------┐     ┌-------------┐ |
+| Release slot | <-- | Release slot | <-- |   Read data  | <-- | Run program | |
+|  (program)   |     |    (data)    |     |     slot     |     └-------------┘ |
+└--------------┘     └--------------┘     └--------------┘                     |
+                                                  |                            |
+                                                  └----------------------------┘
```

Comment:
Should actually describe how these relate to the rest of the document. 

Mark which are XDMA and mark which are commands (preferably with some kind of link or reference to the command below).

I think a paragraph or two here describing the diagram would also be good.